src/python/google_cloud_utils/storage.py

```diff
@@ -188,14 +188,26 @@ def list_blobs(self, remote_path):
     bucket = client.bucket(bucket_name)
     properties = {}
 
-    for blob in bucket.list_blobs(prefix=path):
+    if recursive:
+      delimiter = None
+    else:
+      delimiter = '/'
+
+    iterator = bucket.list_blobs(prefix=path, delimiter=delimiter)
+    for blob in iterator:
       properties['bucket'] = bucket_name
       properties['name'] = blob.name
       properties['updated'] = blob.updated
       properties['size'] = blob.size
 
       yield properties
 
+    if not recursive:
```

Comment:
Can you explain this block more in a comment, is it for dirs ?

Reply:
added a comment.